core/src/main/java/com/ritense/valtimo/autoconfigure/AuthenticationAutoConfiguration.java

```diff
@@ -21,14 +21,17 @@
 import com.ritense.valtimo.security.jwt.authentication.TokenAuthenticationService;
 import com.ritense.valtimo.security.jwt.provider.SecretKeyResolver;
 import java.util.List;
+import com.ritense.valtimo.contract.security.config.oauth2.NoOAuth2ClientsConfiguredCondition;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
 
 @AutoConfiguration
 public class AuthenticationAutoConfiguration {
 
     @Bean
+    @Conditional(NoOAuth2ClientsConfiguredCondition.class)
```

Comment:
Does the application still start/work then this bean is being skipped?

Reply:
Yes. It tested it thoroughly on my local machine.

Basically, the tokenAuthenticationService together with some other Valtimo services were needed to handle the authentication. This can now all be replaced by OAuth 2. When an application is now configured for OAuth 2, it won't need those beans anymore.